### PR TITLE
ocamlPackages.ocaml-protoc: init at 1.2.0

### PR DESCRIPTION
**Pull request by vyorkin**
###### Motivation for this change

Adds [ocaml-protoc](https://github.com/mransan/ocaml-protoc) -- A Protobuf Compiler for OCaml.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

I had to use the `installPhase` attribute to replicate some of the installation steps mentioned it the `protoc`'s Makefile because the make install is broken for 1.2.0 (for example, there is no `lib.install.byte` in the Makefile that is mentioned here https://github.com/mransan/ocaml-protoc/blob/60d2d4dd55f73830e1bed603cc44d3420430632c/Makefile#L133)

### CODE REVIEW

File: pkgs/development/ocaml-modules/protoc/default.nix [hunk 1]
```diff
@@ -0,0 +1,35 @@
+{ stdenv, fetchFromGitHub, buildOcaml, ppx_deriving_protobuf }:
+
+buildOcaml rec {
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/protoc/default.nix]: How did you find out about `buildOcaml`? Please use `mkDerivation`.

File: pkgs/development/ocaml-modules/protoc/default.nix [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  name = "protoc";
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/protoc/default.nix]: The library name is `ocaml-protoc`. So the derivation name should be `ocaml${ocaml.version}-ocaml-protoc-${version}`.

File: pkgs/development/ocaml-modules/protoc/default.nix [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  installPhase = ''
+    mkdir -p tmp/bin
+    export PREFIX=`pwd`/tmp
+    make all.install.build
+    make check_install
+    make lib.install
+    make bin.install
+  '';
+
+  buildInputs = [ ppx_deriving_protobuf ];
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/protoc/default.nix]: Should this be propagated?

File: pkgs/top-level/ocaml-packages.nix [hunk 4]
```diff
@@ -715,6 +715,8 @@ let
 
     process = callPackage ../development/ocaml-modules/process { };
 
+    protoc = callPackage ../development/ocaml-modules/protoc { };
```
Comment (vbgl) [inline pkgs/top-level/ocaml-packages.nix]: The attribute and the directory names should be `ocaml-protoc`.

**Review by vbgl — CHANGES_REQUESTED** [review]

File: pkgs/development/ocaml-modules/protoc/default.nix (hunk 1 above)
Comment (vyorkin) [reply pkgs/development/ocaml-modules/protoc/default.nix]: I've seen that some packages use [buildOcaml](https://github.com/NixOS/nixpkgs/blob/master/pkgs/build-support/ocaml/default.nix), for example [textutils](https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/ocaml-modules/textutils/default.nix)

File: pkgs/top-level/ocaml-packages.nix [hunk 5]
```diff
@@ -715,6 +715,8 @@ let
 
     process = callPackage ../development/ocaml-modules/process { };
 
+    ocaml-protoc = callPackage ../development/ocaml-modules/ocaml-protoc { };
```
Comment (vbgl) [inline pkgs/top-level/ocaml-packages.nix]: Please keep this list sorted.

**Comment by vbgl** [comment]
@GrahamcOfBorg build ocamlPackages.protoc

File: pkgs/top-level/ocaml-packages.nix (hunk 5 above)
Comment (vyorkin) [reply pkgs/top-level/ocaml-packages.nix]: oh, I've forgotten to update after changing the `protoc`->`ocaml-protoc`

**Review by vbgl — APPROVED** [review]

File: pkgs/development/ocaml-modules/protoc/default.nix (hunk 1 above)
Comment (vbgl) [reply pkgs/development/ocaml-modules/protoc/default.nix]: Don’t use `buildOcaml`, unless you want to package a legacy library from Jane Street at version 113.33 for OCaml 4.02.

**Pull request merged**